### DISCUSSION TITLE
Text showing as blocks

### DISCUSSION THREAD

**Discussion by cy33hc** [ocornut/imgui#5985] [New Users Build/Link/Run issues ONLY!]
Hi. I'm trying to use imgui in my app on PS4 using OpenOrbis SDL port. The text is displayed as blocks. Do you know what would be the cause of this? https://github.com/PacBrew/SDL

I'm using the imgui_impl_sdl and  imgui_impl_sdlrenderer backend using latest version from master branch.

![PS4 FTP Client_20221211014903](https://user-images.githubusercontent.com/4947747/206897191-6e110368-116e-42b6-ba3e-0ec09932a1e3.jpg)

**Comment by ocornut** [comment]
See FAQ
https://github.com/ocornut/imgui/blob/master/docs/FAQ.md#q-i-integrated-dear-imgui-in-my-engine-and-little-squares-are-showing-instead-of-text

**Reply by cy33hc** [reply]
Hi. I fixed it.

I changed the Texture creation in the method ImGui_ImplSDLRenderer_CreateFontsTexture() 

FROM:
```
    bd->FontTexture = SDL_CreateTexture(bd->SDLRenderer, SDL_PIXELFORMAT_ABGR8888, SDL_TEXTUREACCESS_STATIC, width, height);
    if (bd->FontTexture == NULL)
    {
        SDL_Log("error creating texture");
        return false;
    }
    SDL_UpdateTexture(bd->FontTexture, NULL, pixels, 4 * width);
    SDL_SetTextureBlendMode(bd->FontTexture, SDL_BLENDMODE_BLEND);
    SDL_SetTextureScaleMode(bd->FontTexture, SDL_ScaleModeLinear);
```

TO:

```
    SDL_Surface* surface = SDL_CreateRGBSurfaceFrom((void*) pixels, width, height, 32, 4 * width, 0xFF000000, 0x00FF0000, 0x0000FF00, 0x000000FF);
    bd->FontTexture = SDL_CreateTextureFromSurface(bd->SDLRenderer, surface);
    if (bd->FontTexture == NULL)
    {
        SDL_Log("error creating texture");
        return false;
    }
    SDL_UpdateTexture(bd->FontTexture, NULL, pixels, 4 * width);
    SDL_SetTextureBlendMode(bd->FontTexture, SDL_BLENDMODE_BLEND);
    SDL_SetTextureScaleMode(bd->FontTexture, SDL_ScaleModeLinear);
    SDL_FreeSurface(surface);
```
